cilium/bpf.cc

```diff
@@ -87,17 +89,24 @@ bool Bpf::open(const std::string& path) {
   } else if (ret.errno_ == ENOENT && log_on_error) {
     ENVOY_LOG(debug, "cilium.bpf_metadata: bpf syscall for map {} failed: {}", path,
               Envoy::errorDetails(ret.errno_));
+    errno = ret.errno_;
   } else if (log_on_error) {
     ENVOY_LOG(warn, "cilium.bpf_metadata: bpf syscall for map {} failed: {}", path,
               Envoy::errorDetails(ret.errno_));
+    errno = ret.errno_;
   }
 
```

Comment:
It seems to me that since the if blocks are conditioned on `log_on_error`, we are missing the error cases where `log_on_error == false`. Maybe simply set the errno here, just before `return false;`?

Reply:
ouch right. fixed. thanks